ios/brave-ios/Sources/Brave/Frontend/UserContent/UserScripts/Scripts_Dynamic/Scripts/Sandboxed/BraveTranslateScript.js

```diff
@@ -0,0 +1,295 @@
+          "url": this[urlProperty],
+          "user": this[userProperty] ?? "",
+          "password": this[passwordProperty] ?? "",
+          "headers": this[requestHeadersProperty] ?? {},
+          "body": body ?? ""
+        }).then((result) => {
+          
+          Object.defineProperties(this, {
+            readyState: { value: XMLHttpRequest.DONE }  // DONE (4)
+          });
+          
+          if (result.value) {
+            Object.defineProperties(this, {
+              status: { value: result.value.statusCode }
+            });
+            
+            Object.defineProperties(this, {
+              statusText: { value: "OK" }
+            });
+            
+            Object.defineProperties(this, {
+              responseType: { value: result.value.responseType }
+            });
+            
+            Object.defineProperty(this, 'response', { writable: true });
+            Object.defineProperty(this, 'responseText', { writable: true });
+            this.responseText = result.value.response;
+            
+            switch (result.value.responseType) {
+              case "arraybuffer": this.response = new ArrayBuffer(result.value.response);
+              case "json": this.response = JSON.parse(result.value.response);
+              case "text": this.response = result.value.response;
+              case "": this.response = result.value.response;
+            }
+          }
+          
+          this.dispatchEvent(new ProgressEvent('loadstart'));
+          this.dispatchEvent(new ProgressEvent(result.error ? 'error' : 'load'));
+          this.dispatchEvent(new ProgressEvent('readystatechange'));
+          this.dispatchEvent(new ProgressEvent('loadend'));
+        });
+        
+        window.webkit = t
+      } catch (e) {
+        return this.realSend(body);
+      }
+    };
+  } else {
+    if (typeof XMLHttpRequest.prototype.realOpen === 'undefined') {
+      XMLHttpRequest.prototype.realOpen = XMLHttpRequest.prototype.open;
+      XMLHttpRequest.prototype.open = function(method, url, async=true, user='', password='') {
+        this.realOpen(method, rewriteUrl(url), async, user, password);
+      }
+    }
+  }
+  ;
+  // An overridden version of onLoadJavascript from translate.js, that fetches
+  // and evaluates secondary scripts (i.e. main.js).
+  // The differences:
+  // 1. change url via rewriteUrl();
+  // 2. process the loaded code via processJavascript().
+  cr.googleTranslate.onLoadJavascript = function(url) {
+      const xhr = new XMLHttpRequest();
+      xhr.open('GET', rewriteUrl(url), true);
+      xhr.onreadystatechange = function() {
+          if (this.readyState !== this.DONE) {
+              return;
+          }
+          if (this.status !== 200) {
+              errorCode = ERROR['SCRIPT_LOAD_ERROR'];
+              return;
+          }
+
+          // nosemgrep
+          new Function(processJavascript(this.responseText)).call(window);
+      }
+      ;
```

Comment:
are these hanging semi-colons for something?

Reply:
Google JS formatter was doing that. I removed all of them.
All Chromium JS also has the same thing lol.